life_expectancy/cleaning.py

```diff
@@ -1,32 +1,73 @@
 import argparse
+from abc import ABC, abstractmethod
 import pandas as pd
 from life_expectancy.i_o import load_data, save_data
+from life_expectancy.countries import Country
 
-def clean_data(df):
+class FileCleaner(ABC):
+    """
+    Abstract class for reading different types of files
+    """
+    ext=''
+    @abstractmethod
+    def clean_data(self, df: pd.DataFrame):
+        """
+        Abstract method for reading different types of files
+        """
+        pass
```

Comment:
**suggestion**: Another option, instead of the ABC, is to use a Callable type alias and functions. There's a whole debate on whether classes with a single public method are really classes (of course they are!) or just functions in disguise. This is the [explored here](https://youtu.be/n2b_Cxh20Fw?t=981), on Arjan's video

Reply:
Probably not worth it to change the code here. But nice resource for other usecases.